odoo/tools/config.py

```diff
@@ -61,61 +68,42 @@ def _deduplicate_loggers(loggers):
         for logger, level in dict(it.split(':') for it in loggers).items()
     )
 
-class configmanager(object):
-    def __init__(self, fname=None):
-        """Constructor.
-
-        :param fname: a shortcut allowing to instantiate :class:`configmanager`
-                      from Python code without resorting to environment
-                      variable
-        """
-        # Options not exposed on the command line. Command line options will be added
-        # from optparse's parser.
-        self.options = {
-            'admin_passwd': 'admin',
-            'csv_internal_sep': ',',
-            'publisher_warranty_url': 'http://services.openerp.com/publisher-warranty/',
-            'reportgz': False,
-            'root_path': None,
-            'websocket_keep_alive_timeout': 3600,
-            'websocket_rate_limit_burst': 10,
-            'websocket_rate_limit_delay': 0.2,
-        }
-
-        # Not exposed in the configuration file.
-        self.blacklist_for_save = set([
-            'publisher_warranty_url', 'load_language', 'root_path',
-            'init', 'save', 'config', 'update', 'stop_after_init', 'dev_mode', 'shell_interface',
-            'longpolling_port',
-        ])
+class configmanager:
+    def __init__(self):
+        self.options = {}
 
         # dictionary mapping option destination (keys in self.options) to MyOptions.
-        self.casts = {}
+        self.options_index = {}
+        self.casts = self.options_index  # deprecated
 
-        self.misc = {}
-        self.config_file = fname
+        self.misc = {}  # deprecated since 17.0
 
         self._LOGLEVELS = dict([
             (getattr(loglevels, 'LOG_%s' % x), getattr(logging, x))
             for x in ('CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET')
         ])
+        self.parser = None  # _build_cli()
+
 
+    def _build_cli(self):
         version = "%s %s" % (release.description, release.version)
         self.parser = parser = optparse.OptionParser(version=version, option_class=MyOption)
 
         # Server startup config
         group = optparse.OptionGroup(parser, "Common options")
-        group.add_option("-c", "--config", dest="config", help="specify alternate config file")
-        group.add_option("-s", "--save", action="store_true", dest="save", default=False,
-                          help="save configuration to ~/.odoorc (or to ~/.openerp_serverrc if it exists)")
-        group.add_option("-i", "--init", dest="init", help="install one or more modules (comma-separated list, use \"all\" for all modules), requires -d")
-        group.add_option("-u", "--update", dest="update",
-                          help="update one or more modules (comma-separated list, use \"all\" for all modules). Requires -d.")
+        group.add_option("-c", "--config", dest="config", file_loadable=False, file_exportable=False,
```

Comment:
looks like if file_loadable=False, file_exportable should alway be False.

It can make sense to be file_exportable=False, file_loadable=True in some case so that we can write a value manually in the config, but not file_loadable=False, file_exportable=True, since even if the value is exported it won't be loaded.


We can have three cases IMHO

-> cannot be loaded or exported : -c, it makes no sence) 
-> can be loaded and exported: port, ... basic config
-> cannot be exported (but loaded): most action generating a side effect or linked to a side effect (-u, -i, ...) . If you want to hard code them in the config, it will work.

Reply:
done, it now raises an error for the invalid 4th case